orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v1schema/V1SchemaExecutionGenerator.java

```diff
@@ -155,4 +153,39 @@ private void addTriggers(
               .orElse(Collections.emptyList()));
     }
   }
+
+  private void addExpectedArtifacts(
+      Map<String, Object> pipeline,
+      PipelineTemplate template,
+      TemplateConfiguration configuration,
+      TemplatedPipelineRequest request) {
+
+    List<HashMap<String, Object>> mergedExpectedArtifacts =
+        mergeExpectedArtifact(
+            emptyListIfNull(configuration.getConfiguration().getExpectedArtifacts()),
+            emptyListIfNull(request.getExpectedArtifacts()));
+
+    if (configuration.getConfiguration().getInherit().contains("expectedArtifacts")) {
+      pipeline.put(
+          "expectedArtifacts",
+          mergeExpectedArtifact(
+              emptyListIfNull(template.getConfiguration().getExpectedArtifacts()),
+              mergedExpectedArtifacts));
+    } else {
+      pipeline.put("expectedArtifacts", mergedExpectedArtifacts);
+    }
+  }
+
+  private List<HashMap<String, Object>> emptyListIfNull(List<HashMap<String, Object>> eas) {
+    return Optional.ofNullable(eas).orElse(Collections.emptyList());
+  }
+
+  private List<HashMap<String, Object>> mergeExpectedArtifact(
```

Comment:
just confirming this is the intent (which, looking at that usages, I think it is) if there are duplicates, the artifacts in the `eas2` will "win", right?

Reply:
yes, that is my intent, this way am spinnaker configuration will be able to override an expected artifact in a template